Microsoft.Toolkit.Uwp.UI.Controls/QuickReturnHeader/QuickReturnHeader.cs

```diff
@@ -0,0 +1,204 @@
+            {
+                me.StopAnimation();
+            }
+        }
+
+        private void QuickReturnHeader_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            if (TargetListView != null)
+            {
+                if (IsQuickReturnEnabled)
+                {
+                    StartAnimation();
+                }
+            }
+        }
+
+        private void ScrollViewer_ViewChanged(object sender, ScrollViewerViewChangedEventArgs e)
+        {
+            if (animationProperties != null)
+            {
+                float oldOffsetY = 0.0f;
+                animationProperties.TryGetScalar("OffsetY", out oldOffsetY);
+
+                var delta = scrollViewer.VerticalOffset - previousVerticalScrollOffset;
+                previousVerticalScrollOffset = scrollViewer.VerticalOffset;
+
+                var newOffsetY = oldOffsetY - (float)delta;
+
+                // Keep values within negativ header size and 0
+                FrameworkElement header = (FrameworkElement)TargetListView.Header;
+                newOffsetY = Math.Max((float)-header.ActualHeight, newOffsetY);
+                newOffsetY = Math.Min(0, newOffsetY);
+
+                if (oldOffsetY != newOffsetY)
+                {
+                    animationProperties.InsertScalar("OffsetY", newOffsetY);
+                }
+            }
+        }
+
+        private void StartAnimation()
+        {
+            if (scrollProperties == null)
+            {
+                scrollProperties = ElementCompositionPreview.GetScrollViewerManipulationPropertySet(scrollViewer);
+            }
+
+            var compositor = scrollProperties.Compositor;
+
+            if (animationProperties == null)
+            {
+                animationProperties = compositor.CreatePropertySet();
+                animationProperties.InsertScalar("OffsetY", 0.0f);
+            }
+
+            var expressionAnimation = compositor.CreateExpressionAnimation("Floor(animationProperties.OffsetY - ScrollingProperties.Translation.Y)");
+
+            expressionAnimation.SetReferenceParameter("ScrollingProperties", scrollProperties);
+            expressionAnimation.SetReferenceParameter("animationProperties", animationProperties);
+
+            headerVisual = ElementCompositionPreview.GetElementVisual((UIElement)TargetListView.Header);
+
+            if (headerVisual != null && IsQuickReturnEnabled)
+            {
+                headerVisual.StartAnimation("Offset.Y", expressionAnimation);
+            }
+        }
+
+        private void StopAnimation()
+        {
+            if (headerVisual != null)
+            {
+                headerVisual.StopAnimation("Offset.Y");
+                animationProperties.InsertScalar("OffsetY", 0.0f);
+
+                var offset = headerVisual.Offset;
+                offset.Y = 0.0f;
+                headerVisual.Offset = offset;
+            }
+        }
+
+        private ScrollViewer scrollViewer;
```

Comment:
Private fields should have an underscore at the start. They should also be at the top of the class.